src/gt4py/next/ffront/func_to_foast.py

```diff
@@ -144,12 +143,11 @@ class FieldOperatorParser(DialectParser[foast.FunctionDefinition]):
     """
 
     @classmethod
-    def _preprocess_definition_ast(cls, definition_ast: ast.AST) -> ast.AST:
-        sta = StringifyAnnotationsPass.apply(definition_ast)
-        ssa = SingleStaticAssignPass.apply(sta)
-        sat = SingleAssignTargetPass.apply(ssa)
-        ucc = UnchainComparesPass.apply(sat)
-        return ucc
+    def _preprocess_definition_ast(cls, ast: ast.AST) -> ast.AST:
+        ast = StringifyAnnotationsPass.apply(ast)
+        ast = SingleStaticAssignPass.apply(ast)
+        ast = SingleAssignTargetPass.apply(ast)
```

Comment:
removed the unchain from python ast transformation, needs to be applied in foast_to_gtir.

Reply:
Actually it doesn't make sense to support unchaining, because also for
`where(0 < field < N,...)` embedded would evaluate to `(0 < field) and (field<N)` instead of `&`.